torch/testing/_internal/opinfo_helper.py

```diff
@@ -0,0 +1,145 @@
+import collections
+import warnings
+from functools import partial
+
+import torch
+from torch.testing._internal.common_cuda import (TEST_CUDA)
+from torch.testing._core import _dispatch_dtypes
+from torch.testing import (all_types_and_complex_and,
+                           all_types_and_complex,
+                           all_types_and_half,
+                           all_types,
+                           complex_types,
+                           floating_and_complex_types,
+                           floating_types_and_half,
+                           floating_types,
+                           integral_types,
+                           floating_types_and,
+                           floating_and_complex_types_and,
+                           integral_types_and,
+                           all_types_and,
+                           )
+
+COMPLETE_DTYPES_DISPATCH = (
+    all_types,
+    all_types_and_complex,
+    all_types_and_half,
+    floating_types,
+    floating_and_complex_types,
+    floating_types_and_half,
+    integral_types,
+    complex_types,
+)
+
+EXTENSIBLE_DTYPE_DISPATCH = (
+    all_types_and_complex_and,
+    floating_types_and,
+    floating_and_complex_types_and,
+    integral_types_and,
+    all_types_and,
+)
+
+# Better way to acquire devices?
+DEVICES = ['cpu'] + (['cuda'] if TEST_CUDA else [])
+
+# Class to tag the dynamically generated types.
+
+
+class _dynamic_dispatch_dtypes(_dispatch_dtypes):
+    pass
+
+
+def get_supported_dtypes(op, sample_inputs_fn, device_type):
```

Comment:
What do you think about taking **kwargs and forwarding them to the `sample_inputs_fn`?

Reply:
What use case do you have in mind? 
Generally we apply partial to `sample_inputs_fn` directly which is consistent when we add `sample_inputs_fn` to OpInfo.

Allowing to pass `kwargs` to `sample_inputs_fn` from here seems a bit confusing.

Reply:
Aha! Good point. Yes, I agree that getting people to partial the function is better, you're right.